test/unit/error.test.ts

```diff
@@ -534,4 +537,83 @@ describe('MongoErrors', () => {
       }
     });
   });
+
+  describe('isResumableError()', () => {
+    it('should return false for errors that are not MongoError', () => {
+      expect(isResumableError(new Error('ah!'))).to.be.false;
+      expect(isResumableError(new TypeError('ah!'))).to.be.false;
+    });
+
+    it('should return true for MongoNetworkError', () => {
+      expect(isResumableError(new MongoNetworkError('ah!'))).to.be.true;
+    });
+
+    it('should return false for an error that is not a MongoError regardless of code property or wire version', () => {
+      // a plain error with and without wire version argument
+      expect(isResumableError(new Error('ah!'))).to.be.false;
+      expect(isResumableError(new Error('ah!'), 9)).to.be.false;
+      expect(isResumableError(new Error('ah!'), 8)).to.be.false;
+
+      const errorWithCode = new (class extends Error {
+        get code() {
+          throw new Error('code on a non-MongoError should not be inspected');
+        }
+        hasErrorLabel() {
+          throw new Error('hasErrorLabel should not be checked on a non-MongoError');
+        }
+      })();
+      // Expectations that prove this syntax provides what is desired for the test
+      expect(errorWithCode).to.be.instanceOf(Error);
+      expect(errorWithCode).to.not.be.instanceOf(MongoError);
+      // Testing that even coded and labeled non-MongoErrors will not get resumed
+      expect(isResumableError(errorWithCode)).to.be.false;
+      expect(isResumableError(errorWithCode, 8)).to.be.false;
+      expect(isResumableError(errorWithCode, 9)).to.be.false;
+    });
+
+    it('should return false for nullish error argument regardless of wire version', () => {
+      expect(isResumableError()).to.be.false;
+      expect(isResumableError(null)).to.be.false;
+      expect(isResumableError(null, null)).to.be.false;
+    });
+
+    it('should return true for labelless MongoError with CursorNotFound code regardless of wire version', () => {
+      const mongoError = new MongoError('ah!');
+      mongoError.code = MONGODB_ERROR_CODES.CursorNotFound;
+      expect(isResumableError(mongoError)).to.be.true;
+      expect(isResumableError(mongoError, 9)).to.be.true;
+      expect(isResumableError(mongoError, 8)).to.be.true;
+    });
+
+    it('should return true for labeled MongoError only if the wireVersion is at least 9', () => {
```

Comment:
can we group the false assertions separately from the true ones here since it's kind of two test cases in one?